### PR TITLE
API to download ERI calculation results and overall system information

### PR DESCRIPTION
**Pull request by urysegal**

### CODE REVIEW

File: examples/water-sto3g-integrals.c [hunk 1]
```diff
@@ -6,100 +6,117 @@
...
+        fprintf(stderr, "Error waiting for job to finish : %s", error.error_message);
+    } else {
 
-    WQC *handler = wqc_init();
+        // Make another call on the same handler, to get information about the ERIs.
+        wqc_reset(handler);
+        res = wqc_get_integrals_details(handler);
 
-    bool rv = calculate_integrals(handler);
+        if (res) {
```
Comment (abcastro) [inline examples/water-sto3g-integrals.c:72]: are you doing
```
if (...) {
```
or
```
if (...)
{
```
?

If I were you I would configure `clang-format`.

File: examples/water-sto3g-integrals.c [hunk 2]
```diff
@@ -6,100 +6,117 @@
...
-    {
-        fprintf(stderr, "Usage: %s output-filename\n", argv[0]);
-        exit(0);
-    }
-}
-
-bool
-calculate_integrals(WQC *job_handler)
+int
+main()
```
Comment (abcastro) [inline examples/water-sto3g-integrals.c:15]: this function is more than 20 lines

Comment (abcastro) [reply examples/water-sto3g-integrals.c:15]: I'm just not sure if "examples" are exceptions.

File: include/webqc-errors.h [hunk 3]
```diff
@@ -12,6 +12,8 @@ typedef struct webqc_handler_t WQC;
 #define WEBQC_NOT_IMPLEMENTED (3) ///< The operation is not yet implemented
 #define WEBQC_OUT_OF_MEMORY (4) ///< Run out of memory
 #define WEBQC_WEB_CALL_ERROR (5) ///< Error calling a web service
+#define WEBQC_NOT_FETCHED (6) ///< A value called for was not yet fetched from the WQC server
+#define WEBQC_IO_ERROR (7) ///< A Some file-related error
```
Comment (abcastro) [inline include/webqc-errors.h]: why not enum?

File: include/webqc-json.h [hunk 4]
```diff
@@ -7,6 +7,38 @@
...
+enum json_field_types {
+    WQC_JSON_INT = 0, /// Integer field
+    WQC_JSON_STRING = 1,  /// String field
+    WQC_JSON_BOOL = 2, /// Boolean field
+    WQC_JSON_ARRAY = 3, /// Array field
+    WQC_JSON_NUMBER = 4 // Number field
+};
+
+/// Instructions for extracting fields from a JSON, used to extract values from HTTP replies.
+struct json_field_info {
```
Comment (abcastro) [inline include/webqc-json.h:21]: no default values here?

Comment (abcastro) [reply include/webqc-json.h:21]: because C?

File: src/web_access.c [hunk 5]
```diff
@@ -310,8 +312,41 @@ bool prepare_get_parameter( WQC *handler, const char *param_name, const char *pa
     return rv;
 }
 
+bool wqc_download_file(WQC *handler, const char *URL, FILE *fp)
+{
+    bool rv = false;
+    CURL *curl = curl_easy_init();
+    CURLcode res;
```
Comment (abcastro) [inline src/web_access.c]: init?

File: src/webqc-eri.c [hunk 6]
```diff
@@ -0,0 +1,332 @@
...
+            shell_index[i] = handler->eri_info.number_of_shells;
+        } else {
+            shell_index[i] = handler->eri_info.basis_functions[(*func_range)[i]].shell_index;
+        }
+    }
+}
+
+
+bool
+wqc_fetch_ERI_values(WQC *handler, const eri_index_t *eri_range_begin, const eri_index_t *eri_range_end)
```
Comment (abcastro) [inline src/webqc-eri.c:209]: Too many lines. Mostly because of the `snprintf`

File: src/webqc-eri.c [hunk 7]
```diff
@@ -0,0 +1,332 @@
...
+        if (fread(handler->eri_info.eri_values.eri_values, sizeof(double), no_of_values, fp) != no_of_values) {
+            wqc_set_error_with_message(handler, WEBQC_IO_ERROR, "Cannot read ERI values from downloaded file");
+            rv = false;
+        }
+    }
+    return rv;
+}
+
+
+// {"begin": [0, 0, 0, 0], "end": [5, 0, 0, 0], "raw_data_url": "https://cloudcompchem.s3.amazonaws.com/public/c08d1e03-4982-4f15-bc99-d4a11bb98e55_56120fe7-16f6-44e9-9fc8-2b4abe34ef25_0_0_0_0_5_0_0_0.eri"}
```
Comment (abcastro) [inline src/webqc-eri.c]: ?

File: src/webqc-eri.c [hunk 8]
```diff
@@ -0,0 +1,332 @@
...
+                rv = read_ERI_values_from_file(handler, fp, begin_shell_index, end_shell_index);
+            }
+        }
+        fclose(fp);
+    }
+    return rv;
+}
+
+
+bool update_eri_values(WQC *handler)
```
Comment (abcastro) [inline src/webqc-eri.c:308]: too many lines

File: examples/water-sto3g-integrals.c (hunk 2 above)
Comment (urysegal) [reply examples/water-sto3g-integrals.c:15]: yes examples are exceptions. they are for educational purpose

File: include/webqc-json.h (hunk 4 above)
Comment (urysegal) [reply include/webqc-json.h:21]: yes, C ....

File: include/webqc-errors.h (hunk 3 above)
Comment (urysegal) [reply include/webqc-errors.h]: made enum

File: src/web_access.c (hunk 5 above)
Comment (urysegal) [reply src/web_access.c]: Fixed

File: src/webqc-eri.c (hunk 7 above)
Comment (urysegal) [reply src/webqc-eri.c]: fixed

File: src/webqc-eri.c (hunk 6 above)
Comment (urysegal) [reply src/webqc-eri.c:209]: slip to two.

File: src/webqc-eri.c (hunk 8 above)
Comment (urysegal) [reply src/webqc-eri.c:308]: split to two

**Pull request merged**